src/wasm/build.sh

```diff
@@ -1,38 +1,21 @@
 #!/bin/bash
 
-set -e
-
 invoke() {
-  em++ -s "EXPORTED_FUNCTIONS=['_malloc', '_free']" -std=c++11 -O3 test-particle.cc -o $1
+  # export EMCC_DEBUG=1 for debug info
+  em++ -s EMIT_EMSCRIPTEN_METADATA -std=c++17 -O3 test-particle.cc -o $1
 }
 
-echo "[Extracting config]"
-invoke wrapper.js
-
-BASE=$(grep -o "DYNAMIC_BASE=[0-9]\+" wrapper.js)
-TOP=$(grep -o "DYNAMICTOP_PTR=[0-9]\+" wrapper.js)
-MEM=$(grep -o 'env\["__memory_base"\]=[0-9]\+' wrapper.js | grep -o "[0-9]*")
-
-CTMP=cfg.tmp
-cat <<EOF | tee $CTMP
-// tslint:disable
-// Temporarily generated by the wasm build process (but needs to be checked into src)
-export const WasmConfig = {
-  MEMORY_BASE: $MEM,
-  ${BASE/=/: },
-  ${TOP/=/: }
-};
-EOF
-
-CFG=../runtime/wasm-config.ts
-diff $CTMP $CFG || cp -v $CTMP $CFG
-
-rm wrapper.* $CTMP
+if [[ $1 = -g ]]; then
+  # Generate the glue code (useful as a development reference).
+  invoke glue.js
+  rm glue.wasm
+  which js-beautify >/dev/null && js-beautify -r glue.js 
+  exit
+fi
 
-echo -e "\n[Building module]"
 invoke output.wasm
 
 if which wasm2wat >/dev/null; then
-  wasm2wat output.wasm > output.wat
+  wasm2wat output.wasm -o output.wat
   grep "^  (import" output.wat
```

Comment:
What's this checking?

Reply:
It's displaying the import table for me. Just a convenience while I'm working on the interface between js & wasm.